test/unit/math/prim/fun/generalized_inverse_test.cpp

```diff
@@ -0,0 +1,24 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathMatrixPrim, Zero) {
+  stan::math::matrix_d m0(0, 0);
+  stan::math::matrix_d ginv = stan::math::generalized_inverse(m0);
+  EXPECT_EQ(0, ginv.rows());
+  EXPECT_EQ(0, ginv.cols());
+}
+
+TEST(MathMatrixPrim, Equal) {
+  using stan::math::generalized_inverse;
+
+  stan::math::matrix_d m1(3, 2);
+  m1 << 1, 2, 2, 4, 1, 2;
+
+  stan::math::matrix_d m2(2, 3);
+  m2 << 1 / 30.0, 1 / 15.0, 1 / 30.0, 1 / 15.0, 2 / 15.0, 1 / 15.0;
+  stan::math::matrix_d m3 = generalized_inverse(m1);
+  for (int j = 0; j < m3.cols(); j++)
+    for (int i = 0; i < m3.rows(); i++)
+      EXPECT_NEAR(m2(i, j), m3(i, j), 1e-5);
```

Comment:
Such comparisons are so common, we have a macro for them:
```suggestion
  EXPECT_MATRIX_NEAR(m2, m3, 1e-5);
```
To use it you will need to include appropriate header:
```
#include <test/unit/util.hpp>
```

Reply:
Also does this achieve only 1e-5 accuracy? In tests for most functions we require 1e-9.

Reply:
I did get it to achieve the 1e-9 accuracy. But it depends on the user's tolerance. When Cholesky is successful this method will exactly reproduce the generalized inverse. However, due to numerical instability in Cholesky decompositions it is sometimes necessary to add a small jitter to the diagonal. This is a well known issue but the user should supply the tolerance they are comfortable with.

Reply:
I added the EXPECT_MATRIX_NEAR as well.